crates/wash-lib/src/app.rs

```diff
@@ -419,6 +435,55 @@ pub async fn load_app_manifest(source: AppManifestSource) -> anyhow::Result<AppM
         .context("app manifest loader timed out")?
 }
 
+/// Validate the contents of a manifest file and optionally validate the OCI references
+pub async fn validate_manifest_file(
+    manifest_contents: &str,
+    oci_check: bool,
+) -> Result<(Manifest, Vec<ValidationFailure>)> {
+    let manifest = serde_yaml::from_slice(manifest_contents.as_ref())
+        .context("failed to parse manifest content")?;
```

Comment:
would love a `with_context` here just to make sure we always know *which* file failed